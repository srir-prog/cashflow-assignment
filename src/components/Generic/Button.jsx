import React from "react";
const Button = ({ onClick, buttonIcon, className }) => {
  return (
    <button className={className} onClick={onClick}>
      <i className={buttonIcon}></i>
    </button>
  );
};

export default Button;
