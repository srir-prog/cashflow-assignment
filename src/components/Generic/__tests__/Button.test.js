import React from 'react';
import { shallow } from 'enzyme';
import Button from './../Button';
import renderer from 'react-test-renderer';

describe('Button component', () => {
    let component;
    const props = {
        onClick: jest.fn(),
        buttonIcon: 'buttonIcon',
        className: 'className'
    }
    beforeEach(() => {
        component = shallow(<Button {...props} />);
    });
    it('should render correctly', () => {
        //Snapshot testing.
        const tree = renderer.create(<Button {...props} />)
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('should render <button> tag', () => {
        const buttonElement = component.find('button');
        expect(buttonElement.length).toBe(1);
    });
});
