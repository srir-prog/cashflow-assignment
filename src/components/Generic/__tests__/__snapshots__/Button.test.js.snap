// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<button
  className="className"
  onClick={[MockFunction]}
>
  <i
    className="buttonIcon"
  />
</button>
`;
